pkg/apiserver/debugapi/service.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debugapi
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joomcode/errorx"
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/user"
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/utils"
+)
+
+var (
+	ErrNS                = errorx.NewNamespace("error.api.debugapi")
+	ErrComponentClient   = ErrNS.NewType("invalid_component_client")
+	ErrEndpointConfig    = ErrNS.NewType("invalid_endpoint_config")
+	ErrInvalidStatusPort = ErrNS.NewType("invalid_status_port")
+)
+
+func registerRouter(r *gin.RouterGroup, auth *user.AuthService, s *Service) {
+	endpoint := r.Group("/debugapi")
+	endpoint.Use(auth.MWAuthRequired())
+
+	endpoint.POST("/endpoint", s.RequestEndpoint)
+	endpoint.GET("/endpoints", s.GetEndpointList)
+}
+
+type endpoint struct {
+	EndpointAPIModel
+	Client Client
+}
+
+type Service struct {
+	endpointMap map[string]endpoint
+}
+
+func newService(clientMap *ClientMap) (*Service, error) {
+	s := &Service{endpointMap: map[string]endpoint{}}
+
+	for _, e := range endpointAPIList {
+		client, ok := clientMap.Get(e.Component)
+		if !ok {
+			return nil, ErrComponentClient.New("%s type client not found, id: %s", e.Component, e.ID)
```

Comment:
Should this be panic instead as it seems to be definitely a programming mistake?